repos/spack_repo/builtin/packages/repeatmodeler/package.py

```diff
@@ -47,6 +47,46 @@ class Repeatmodeler(Package):
     depends_on("blat", type="run", when="@2.0.4:")
     depends_on("ltr-retriever", type="run", when="@2.0.4:")
 
+    def patch(self):
+        file_list = [
+            "BuildDatabase",
+            "Job.pm",
+            "LTRPipeline",
+            "MultAln.pm",
+            "NeedlemanWunschGotohAlgorithm.pm",
+            "Refiner",
+            "RepeatClassifier",
+            "RepeatModeler",
+            "RepeatUtil.pm",
+            "RepModelConfig.pm",
+            "SeedAlignmentCollection.pm",
+            "SeedAlignment.pm",
+            "SequenceSimilarityMatrix.pm",
+            "Task.pm",
+            "ThreadedJob.pm",
+            "ThreadedTaskSimple.pm",
+            "TRFMask",
+            "util/alignAndCallConsensus.pl",
+            "util/align.pl",
+            "util/AutoRunBlocker.pl",
+            "util/bestwindow.pl",
+            "util/Blocker.pl",
+            "util/ClusterPartialMatchingSubs.pl",
+            "util/CntSubst",
+            "util/extendFlankingSeqs.pl",
+            "util/fasta-trf-filter.pl",
+            "util/generateSeedAlignments.pl",
+            "util/Linup",
+            "util/renameIds.pl",
+            "util/resolveIndels.pl",
+            "util/rmblast.pl",
+            "util/TSD.pl",
+            "util/viewMSA.pl",
+            "util/visualizeAlignPNG.pl",
+        ]
+        for f in file_list:
+            filter_file("^#!/usr/bin/perl.*$", "#!/usr/bin/env perl", f, ignore_absent=True)
```

Comment:
Some of the files have 

> /usr/local/bin/perl

and this would not work there.

Also, we're dropping the '-w', that's not ideal.

Maybe something like this:
`filter_file("^#!/usr/(?:local/)?bin/perl( -w)?.*$", "#!/usr/bin/env perl${1}", f, ignore_absent=True)`

Although I'm not sure if we can use backrefs here at all and also I'm not sure if the Python regexes support everything Perl would. Maybe this needs to be adapted.

Reply:
I bet this will work fine, I'm dealing with some issues with a dependency so testing is pending that getting resolved.